### PR TITLE
Support for Pancreas-CT dataset

### PR DESCRIPTION
**Pull request by mzuevx**
Added special mode to brats annotation converter and nifti reader to implement 2D processing of Pacnreas-CT data.

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 1]
```diff
@@ -214,11 +214,17 @@ The main difference between this converter and `super_resolution` in data organi
 * `unicode_character_recognition` - converts [Kondate](http://web.tuat.ac.jp/~nakagawa/database/en/kondate_about.html) dataset and [Nakayosi](http://web.tuat.ac.jp/~nakagawa/database/en/about_nakayosi.html) for handwritten Japanese text recognition task , and [SCUT-EPT](https://github.com/HCIILAB/SCUT-EPT_Dataset_Release) for handwritten simplified Chinese text recognition task to `CharacterRecognitionAnnotation`.
   * `annotation_file` - path to annotation file in txt format.
   * `decoding_char_file` - path to decoding_char_file, consisting of all supported characters separated by '\n' in txt format.
-* `brats` - converts BraTS dataset format to `BrainTumorSegmentationAnnotation` format.
+* `brats` - converts BraTS dataset format to `BrainTumorSegmentationAnnotation` format. Also can be used to annotate other nifti-based datasets.
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:217]:
```suggestion
* `brats` - converts BraTS dataset format to `BrainTumorSegmentationAnnotation` format. Also can be used to convert other nifti-based datasets.
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 2]
```diff
@@ -214,11 +214,17 @@ The main difference between this converter and `super_resolution` in data organi
...
   Optionally you can provide relative path for these subdirectories (if they have different location) using `image_folder` and `mask_folder` parameters respectively.
-  * `mask_channels_first` - allows read gt mask nifti files and transpose in order where channels first (Optional, default False)
+  * `mask_channels_first` - allows read gt mask nifti files and transpose in order where channels first (Optional, default `False`)
   * `labels_file` - path to file, which contains labels (optional, if omitted no labels will be shown)
+  * `relaxed_names` - allows to use more relaxed search of labels matching only numeric ids. Optional, by default full name matching required.
+  * `multi_frame` - allows annotation of 3D images as sequence of 2D frames (optional, default `False`)
+  * `frame_separator` - string separator between file name and frame number in `multi_frame` (optional, default `#`)
+  * `frame_axis` - number of frame axis in 3D Image (optional, default `-1`, last axis)
+  * `as_regression` - allows dataset annotation as `NiftiRegressionAnnotation` annotation (optional, default `False`)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:226]:
```suggestion
  * `as_regression` - allows dataset conversion as `NiftiRegressionAnnotation` annotation (optional, default `False`)
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 3]
```diff
@@ -214,11 +214,17 @@ The main difference between this converter and `super_resolution` in data organi
...
-* `brats` - converts BraTS dataset format to `BrainTumorSegmentationAnnotation` format.
+* `brats` - converts BraTS dataset format to `BrainTumorSegmentationAnnotation` format. Also can be used to annotate other nifti-based datasets.
   * `data_dir` - dataset root directory, which contain subdirectories with validation data (`imagesTr`) and ground truth labels (`labelsTr`).
   Optionally you can provide relative path for these subdirectories (if they have different location) using `image_folder` and `mask_folder` parameters respectively.
-  * `mask_channels_first` - allows read gt mask nifti files and transpose in order where channels first (Optional, default False)
+  * `mask_channels_first` - allows read gt mask nifti files and transpose in order where channels first (Optional, default `False`)
   * `labels_file` - path to file, which contains labels (optional, if omitted no labels will be shown)
+  * `relaxed_names` - allows to use more relaxed search of labels matching only numeric ids. Optional, by default full name matching required.
+  * `multi_frame` - allows annotation of 3D images as sequence of 2D frames (optional, default `False`)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:223]:
```suggestion
  * `multi_frame` - allows convertt annotation of 3D images as sequence of 2D frames (optional, default `False`)
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/brats.py [hunk 4]
```diff
@@ -71,13 +104,28 @@ def convert(self, check_content=False, **kwargs):
...
-            annotations.append(annotation)
+            if self.multi_frame:
+                nib_image = nib.load(file_in_dir)
+                total_frames = nib_image.dataobj.shape[self.frame_axis]
+                for frame_cnt in range(total_frames):
+                    annotation = annotation_class(
+                        "{}{}{}".format(str(image_folder / file_name), self.frame_separator, frame_cnt),
+                        "{}{}{}".format(str(mask_folder / mask_file_name), self.frame_separator, frame_cnt),
+                        **reader_config
+                        # loader=GTMaskLoader.NIFTI_CHANNELS_FIRST if self.mask_channels_first else GTMaskLoader.NIFTI
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/brats.py:116]: please remove comments

File: tools/accuracy_checker/accuracy_checker/representation/regression_representation.py [hunk 5]
```diff
@@ -135,3 +135,27 @@ def value(self):
...
+                 multi_frame=False, separator='#', frame_axis=-1):
+        super().__init__(identifier)
+        self.value_file = value_file
+        self._reader_config = {'type': 'nifti_reader', 'to_4D': to_4D, 'channels_first': mask_channels_first,
+                               'multi_frame': multi_frame, 'separator': separator, 'frame_axis': frame_axis}
+        self._value = None
+
+    @property
+    def value(self):
+        # return np.zeros([1,512,512])
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/representation/regression_representation.py:150]: please remove debug comments

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/representation/regression_representation.py]: Fixed

File: tools/accuracy_checker/accuracy_checker/annotation_converters/brats.py (hunk 4 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/brats.py]: Fixed

**Review by eaidova — APPROVED** [review]

**Pull request merged**